compiler/rustc_mir/src/dataflow/framework/engine.rs

```diff
@@ -208,12 +208,19 @@ where
             }
         }
 
+        // `state` is not actually used between iterations;
+        // this is just an optimization to avoid reallocating
+        // every iteration.
         let mut state = analysis.bottom_value(body);
```

Comment:
Here's an idea for a further optimization: change this to

```suggestion
        let mut state = MaybeUninit::uninit();
```

since it's always initialized with `clone_from` below. Might not be worth the unsafety though 🤷